### PR TITLE
add explanation for the service account considerations for jenkins ag…

### PR DESCRIPTION
**Pull request by gabemontero**
…ents and k8s plugin

Fixes https://github.com/openshift/jenkins/issues/484

@openshift/sig-developer-experience @bmcelvee ptal

### CODE REVIEW

**Comment by bparees** [comment]
lgtm

File: using_images/other_images/jenkins.adoc [hunk 1]
```diff
@@ -514,6 +514,33 @@ an entrypoint. For more details about this, refer to the official
 https://wiki.jenkins-ci.org/display/JENKINS/Distributed+builds#Distributedbuilds-Launchslaveagentheadlessly[Jenkins
 documentation].
 
+==== Permission considerations
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: [[permission-considerations]]

s/ Permission considerations/ Permission Considerations

File: using_images/other_images/jenkins.adoc [hunk 2]
```diff
@@ -514,6 +514,33 @@ an entrypoint. For more details about this, refer to the official
...
+
+Consider for a moment in the above ConfigMap example the `<serviceAccount>` element of the
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: For the first paragraph:

In the previous ConfigMap example, the `<serviceAccount>` element of the 
Pod Template XML is the {product-title} Service Account used for the 
resulting Pod. The service account credentials mounted into the Pod, with permissions 
 associated with the service account, control which operations against the {product-title} master
are allowed from the Pod.

File: using_images/other_images/jenkins.adoc [hunk 3]
```diff
@@ -514,6 +514,33 @@ an entrypoint. For more details about this, refer to the official
...
 
+==== Permission considerations
+
+Consider for a moment in the above ConfigMap example the `<serviceAccount>` element of the 
+Pod Template XML.  As the name implies, it is the {product-title} Service Account used for the 
+resulting Pod, where those Service Account credentials mounted into the Pod, with whatever permissions 
+are associated with the ServiceAccount, control which operations against the {product-title} master
+are allowed from the Pod.
+
+Some details to consider with regard to the Service Accounts used for Pod launched by the Kubernetes
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: s/Some details to consider with regard to the Service Accounts used for Pod/Consider the following with service accounts used for the Pod,

File: using_images/other_images/jenkins.adoc [hunk 4]
```diff
@@ -514,6 +514,33 @@ an entrypoint. For more details about this, refer to the official
...
+Plug-in running in the {product-title} Jenkins image:
+
+- If you use the example template for Jenkins provided by {product-title}, the `jenkins` service 
+account is defined with the `edit` role for the project Jenkins is running in, and the master Jenkins
+Pod has that Service Account mounted.
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: s/Service Account/service account

File: using_images/other_images/jenkins.adoc [hunk 5]
```diff
@@ -514,6 +514,33 @@ an entrypoint. For more details about this, refer to the official
...
+- The two default maven and nodejs Pod Templates injected into the Jenkins configuration are also
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: s/maven and nodejs/Maven and Nodejs?

File: using_images/other_images/jenkins.adoc [hunk 6]
```diff
@@ -514,6 +514,33 @@ an entrypoint. For more details about this, refer to the official
...
+set to use the same service account as the master.
+- Any Pod Templates auto-discovered by the link:https://github.com/openshift/jenkins-sync-plugin[OpenShift Sync plug-in]
+as a result of Image streams or Image stream tags having the required label or annotations have 
+their service account set to the master's service account.
+- For the other ways you can provide a Pod Template definition into Jenkins and the Kubernetes plug-in,
+you have to explicitly specify the service account to use. 
+- Those other ways include the Jenkins console, the `podTemplate` pipeline DSL provided by the Kubernetes
+plug-in, or labeling a ConfigMap whose data is the XML configuration for a Pod Template.
+- If you do not specify a value for the service account, the `default` service account is used.
+- You need to ensure that whatever service account is used has the necessary permissions, roles, etc. defined within
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: s/etc./and so on,

**Comment by gabemontero** [comment]
Yep it applies back to 3.7

**Comment by wzheng1** [comment]
LGTM, thanks!

**Comment by bmcelvee** [comment]
Thanks, @wzheng1! 

@gabemontero please squash the commits, then I'll merge and cherry-pick. 

Thanks again!

**Comment by gabemontero** [comment]
commits squashed @bmcelvee

**Pull request merged**